configure

```diff
@@ -992,9 +1004,13 @@ case "${ARCH}" in
                 fi
 
                 if test $buildneon -eq 1; then
-                    CFLAGS="${CFLAGS} ${floatabi} -DARM_NEON_ADLER32"
-                    SFLAGS="${SFLAGS} ${floatabi} -DARM_NEON_ADLER32"
-
+                    if test $MFPU_NEON_AVAILABLE -eq 1;then
+                        CFLAGS="${CFLAGS} ${floatabi} -mfpu=neon -DARM_NEON_ADLER32"
+                        SFLAGS="${SFLAGS} ${floatabi} -mfpu=neon -DARM_NEON_ADLER32"
+                    else
+                        CFLAGS="${CFLAGS} ${floatabi} -DARM_NEON_ADLER32"
+                        SFLAGS="${SFLAGS} ${floatabi} -DARM_NEON_ADLER32"
+                    fi
```

Comment:
Instead of duplicating the add of `${floatabi}` in else block, you should move adding it above the IF block, and adding `-DARM_NEON_ADLER32` below the IF block, so they are done no matter what value $MFPU_NEON_AVAILABLE has... This is consistent how it is done on other architectures. It makes the code a little longer, but a lot more readable.

Reply:
Would it be better to put ${floatabi} and -DARM_NEON_ADLER32 together below IF block?
"
if test $buildneon -eq 1; then
                    if test $MFPU_NEON_AVAILABLE -eq 1;then
                        CFLAGS="${CFLAGS} -mfpu=neon"
                        SFLAGS="${SFLAGS} -mfpu=neon"
                    fi

                    CFLAGS="${CFLAGS} ${floatabi} -DARM_NEON_ADLER32"
                    SFLAGS="${SFLAGS} ${floatabi} -DARM_NEON_ADLER32"

                    ARCH_STATIC_OBJS="${ARCH_STATIC_OBJS} adler32_neon.o"
                    ARCH_SHARED_OBJS="${ARCH_SHARED_OBJS} adler32_neon.lo"
                fi
"

Reply:
I usually don't like changing order of compiler parameters, because the result can be unpredictable. That's why we previously used empty variables instead when something was optional.